### PR TITLE
Improve function names and other minor changes (nfc)

### PR DESCRIPTION
**Pull request by LucasSte**
This PR has general fixes in Solang, namely:

1. Added an assertion at `fn must_fail` to make sure the called function returns an error.
2. Rename `edges` to `successors`, as the function returns the successors of a block.
3. Rename `clone_for_parent_block` to `deep_clone`.
4. Rename `set_maxi_signed` to `get_max_signed`
5. Rename `set_max_unsigned` to `set_max_unsigned`
6. Fix typos in comments and in function names.

Edit: I slipped in another change.
7. Use `.expect("reason")` instead of `if {} else { unreachable!()}` at `dead_storage.rs`.

### CODE REVIEW

File: tests/solana_tests/create_contract.rs [hunk 1]
```diff
@@ -367,7 +367,7 @@ fn missing_contract() {
                 is_signer: false,
             },
         ])
-        .must_fail();
+        .may_fail();
```
Comment (seanyoung) [inline tests/solana_tests/create_contract.rs:370]: Missing check on result, This is exactly why `fn must_fail` should check the result itself

File: tests/solana.rs [hunk 2]
```diff
@@ -1719,7 +1719,7 @@ impl<'a, 'b> VmFunction<'a, 'b> {
         }
     }
 
-    fn must_fail(&mut self) -> ProgramResult {
+    fn may_fail(&mut self) -> ProgramResult {
```
Comment (seanyoung) [inline tests/solana.rs:1722]: This is seems quite error prone. Would it not be better to retain the name `must_fail` and assert that the result is non-zero?

**Review by seanyoung — CHANGES_REQUESTED** [review]
This PR needs a much better title

File: tests/solana.rs (hunk 2 above)
Comment (LucasSte) [reply tests/solana.rs]: Fixed

**Comment by LucasSte** [comment]
I slipped another change in `dead_storage.rs`.

File: tests/solana_tests/primitives.rs [hunk 3]
```diff
@@ -1275,7 +1261,7 @@ fn test_overflow_detect_unsigned() {
             // Calculate a number that when multiplied by first_operand_rand, the result will overflow N bits
             let second_operand_rand = rng.gen_biguint_range(&BigUint::from(2usize), &limit);
 
-            let res = contract
+            let _ = contract
```
Comment (seanyoung) [inline tests/solana_tests/primitives.rs:1264]: Does this and other lines need `let _ = `?

**Review by seanyoung — COMMENTED** [review]
How about "Improve function names and other minor changes (nfc)"

File: tests/solana_tests/primitives.rs (hunk 3 above)
Comment (LucasSte) [reply tests/solana_tests/primitives.rs]: Removed.

**Review by seanyoung — APPROVED** [review]
Very nice, thanks

**Pull request merged**